sale_order_net_price_control/models/sale_order_line.py

```diff
@@ -0,0 +1,71 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, _, fields
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    offer_price = fields.Boolean()
+    admits_offer_prices = fields.Boolean(related='product_id.admits_offer_prices', store=True)
+
+    def _compute_amount(self):
+        super()._compute_amount()
+        if not self.env.user.has_group('sale_order_net_price_control.group_sale_at_any_price'):
+
+            for record in self.filtered(lambda r: r.product_id and r.product_uom_qty > 0):
+
+                net_price_sale = record.price_subtotal / record.product_uom_qty
+
+                purchase_invoice = self.env['account.move.line'].search([
+                    ('product_id', '=', record.product_id.id),
+                    ('quantity', '>', 0),
+                    ('move_id.move_type', '=', 'in_invoice'),
+                    ('move_id.state', '=', 'posted'),
+                    ('display_type', '=', 'product'),
+                ], limit=1, order='name desc')
+
+                if purchase_invoice:
+                    invoice_line = purchase_invoice.filtered(
+                    lambda x: x.product_id.id == record.product_id.id
+                    )
+                    net_price_purchase = invoice_line.price_subtotal / invoice_line.quantity
+
+                    if net_price_sale <= net_price_purchase:
+                        raise ValidationError(_(
+                            "Net price of '%s', '%.1f' must be higher that the last purchase invoice price '%s'"
+                        )%(record.product_template_id.name,net_price_sale,net_price_purchase) )
+                    if not record.offer_price:
+                        if net_price_sale <= self._get_price_from_lowest_pricelist_id():
```

Comment:
There's another `self` that is incorrect, because within this method self is used as a standalone recordset:
```suggestion
                        if net_price_sale <= record._get_price_from_lowest_pricelist_id():
```

In general, please check `_compute_amount` is working in multi mode